data/transactions/verify/txn.go

```diff
@@ -438,11 +452,11 @@ func logicSigVerify(txn *transactions.SignedTxn, groupIndex int, groupCtx *Group
 	pass, cx, err := logic.EvalSignatureFull(groupIndex, &ep)
 	if err != nil {
 		logicErrTotal.Inc(nil)
-		return fmt.Errorf("transaction %v: rejected by logic err=%v", txn.ID(), err)
+		return LogicSigError{groupIndex, fmt.Errorf("transaction %v: rejected by logic err=%v", txn.ID(), err)}
```

Comment:
Do you want to use %w here so it can unwrapped?

Reply:
Good catch